src/power_grid_model/validation/validation.py

```diff
@@ -216,26 +208,6 @@ def validate_unique_ids_across_components(data: SingleDataset) -> list[MultiComp
     return all_cross_unique(data, [(component, "id") for component in data])
 
 
-def validate_ids_exist(update_data: SingleDataset, input_data: SingleDataset) -> list[IdNotInDatasetError]:
-    """
-    Checks if all ids of the components in the update data exist in the input data. This needs to be true, because you
-    can only update existing components.
-
-    This function should be called for every update dataset in a batch set
-
-    Args:
-        update_data: A single update dataset
-        input_data: A power-grid-model input dataset
-
-    Returns:
-        An empty list if all update data ids exist in the input dataset, or a list of IdNotInDatasetErrors for
-        all update components of which the id does not exist in the input dataset
-
-    """
-    errors = (all_ids_exist_in_data_set(update_data, input_data, component, "input_data") for component in update_data)
-    return list(chain(*errors))
```

Comment:
i don't think this rule should be entirely removed. in particular:

* missing IDs are supported iff:
  * the size of the update data set for a specific component has exactly the same size per scenario as the amount of components of that type in the input data set. (this is a rule that we can check)
  * and there are no IDs provided for any of the components of this type in this update data set (this is another rule)
* in any other case: (these rules already exist but may need adjustments)
  * providing IDs is mandatory (this is another rule)
  * and all IDs need to exist (this is another rule)
  * and all IDs need to represent a component of this same component type. (this is another rule)

Reply:
New checks that resemble those in the core will be added under a different name.